File: app/src/main/java/com/hencoder/hencoderpracticedraw2/practice/Practice13ShadowLayerView.java
package com.hencoder.hencoderpracticedraw2.practice;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;

public class Practice13ShadowLayerView extends View {

    private Paint mPaint;

    public Practice13ShadowLayerView(Context context) {
        this(context, null);
    }

    public Practice13ShadowLayerView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public Practice13ShadowLayerView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public void init() {
        mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);

        // 使用 Paint.setShadowLayer() 设置阴影
        mPaint.setShadowLayer(5, 5, 5, Color.RED);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);

        mPaint.setTextSize(120);
        canvas.drawText("Hello HenCoder", 50, 200, mPaint);
    }

}

File: app/src/main/java/com/hencoder/hencoderpracticedraw2/practice/Practice13ShadowLayerView.kt
package com.hencoder.hencoderpracticedraw2.practice

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View

/**
 * Created by xh on 2017/11/1.
 */
class Practice13ShadowLayerView : View {

    val mPaint = Paint(Paint.ANTI_ALIAS_FLAG)

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    private val mLeftMargin = 50f
    private val mTopMargin = 200f

    private val mTextSize = 120f
    private val mText = "Hello HenCoder"

    override fun onDraw(canvas: Canvas?) {
        super.onDraw(canvas)

        mPaint.textSize = mTextSize
        mPaint.setShadowLayer(5f, 5f, 5f, Color.RED)

        canvas!!.drawText(mText, mLeftMargin, mTopMargin, mPaint)


    }

}